### PR TITLE
Fix/disable some notifications when machete file opened & focused

### LINKED ISSUE #1238: Notifications about duplicate/non-existent branch should NOT be displayed when .git/machete is visible (opened & focused) in editor

### PR DESCRIPTION
**Pull request by LukaszKontowski**
Closes #1238

### CODE REVIEW

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java [hunk 1]
```diff
@@ -226,25 +228,41 @@ private void refreshModel(
...
     doOnUIThreadWhenReady.run();
   }
 
+  @UIEffect
+  private boolean macheteFileIsOpened(Path macheteFilePath) {
+    val fileEditorManager = FileEditorManager.getInstance(project);
+    val macheteVirtualFile = List.of(fileEditorManager.getOpenFiles())
+        .find(virtualFile -> virtualFile.getPath().equals(macheteFilePath.toString()));
+    if (macheteVirtualFile.isEmpty()) {
+      return false;
```
Comment (LukaszKontowski) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:260]: such scenario should not happen in reality, but it is a way to safely call `get()` on the optional value

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java [hunk 2]
```diff
@@ -226,25 +228,41 @@ private void refreshModel(
 
     setModel(new GraphTableModel(repositoryGraph));
 
-    Set<String> skippedBranchNames = repositorySnapshot.getSkippedBranchNames();
-    if (skippedBranchNames.nonEmpty()) {
-      val notification = getSkippedBranchesNotification(repositorySnapshot, gitRepository);
-      VcsNotifier.getInstance(project).notify(notification);
-    }
+    if (!macheteFileIsOpened(macheteFilePath)) {
```
Comment (mkondratek) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:231]: I am not convinced of this idea - to show or not to show a notification depending on some file being opened. For me, it looks like a bug.

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java [hunk 3]
```diff
@@ -226,25 +228,41 @@ private void refreshModel(
...
 
     repaint();
     revalidate();
     doOnUIThreadWhenReady.run();
   }
 
+  @UIEffect
+  private boolean macheteFileIsOpened(Path macheteFilePath) {
+    val fileEditorManager = FileEditorManager.getInstance(project);
+    val macheteVirtualFile = List.of(fileEditorManager.getOpenFiles())
```
Comment (mkondratek) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:257]: Out of curiosity... why `getOpenFiles()` returns files not a file? I assume that it does not include files that you can access from the tabs but it does include files opened in the split editor window in particular, right?

What I mean by tabs:
image

What I mean by split editor window:
image

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java (hunk 2 above)
Comment (PawelLipski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:231]: Hmmm the problem is that generally we want to notify the user of duplicate and/or non-existent branches... but it's rather annoying when the notifications pop up during an edit ☹️ compare it to errors during editing code (no notification, just red squiggles) vs during build from IntelliJ (probably some notification-like stuff shows up)

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java (hunk 3 above)
Comment (LukaszKontowski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:257]: `getOpenFiles()` returns all opened files. It returns all files that are opened in the editor tabs.

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java (hunk 2 above)
Comment (LukaszKontowski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:231]: @mkondratek but that is the idea of this issue:
https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1238
> Notifications about duplicate/non-existent branch should NOT be displayed when .git/machete is opened in editor

Comment (PawelLipski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:231]: ☝🏻 Yeah, hence I'm taking the burden of defending the idea 😅

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java (hunk 3 above)
Comment (mkondratek) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:257]: hm, so it's not what we want , right? the notification should be skipped (/disabled) only when the machete file is visible, right?

Comment (LukaszKontowski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:257]: I am not sure. It seemed to me that notifications should not be displayed if the file is opened. @PawelLipski please specify :)

Comment (PawelLipski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:257]: > only when the machete file is visible, right?

Right, +1, my fault. Of course I meant **opened & focused** aka visible, not just opened but unfocused. Lemme fix that on the issue. Sorry for the confusion, pls rework @LukaszKontowski

Comment (LukaszKontowski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java]: fixed :)

**Comment by mkondratek** [comment]
It does not work 😕 the branch remains underlined

image

**Comment by LukaszKontowski** [comment]
> It does not work 😕 the branch remains underlined
> 
> image

Hm, but this PR is not about underlining. My changes from this PR should have such effect, that **notifications** (appearing on the bottom of the screen) will not be displayed when machete file is selected.

**Comment by mkondratek** [comment]
@LukaszKontowski oh, ofc 🤦  nvm

**Review by mkondratek — APPROVED** [review]
lgtm

**Review by PawelLipski — APPROVED** [review]
:benissimo:

**Comment by mkondratek** [comment]
> > It does not work confused the branch remains underlined
> 
> Actually... what was the IDE error on the screenshot? 🤔 @mkondratek

most likely it was not ours. I used to check them every time I saw them but since they started to appear every time I launch the ide with `runIde`, I stopped Xd

and IIRC they were always from intellij (not ours)

**Comment by mkondratek** [comment]
Ive check the props from the `edit custom properties...` (https://plugins.jetbrains.com/docs/intellij/enabling-internal.html) - nothing there. So I assume it's off

**Pull request merged**